coral-component-list/src/scripts/SelectList.js

```diff
@@ -253,6 +253,10 @@ class SelectList extends BaseComponent(HTMLElement) {
 
     const item = event.matchedTarget;
     this._toggleItemSelection(item);
+    if (item.hasAttribute("href")) {
+    const href = item.getAttribute("href");
+    window.open(href, '_self');
+    }
```

Comment:
Please try to implement this functionality inside component only 
try listening to event : **coral-selectlist:change**